### PR TITLE
Expose Strawberry info in resolvers

### PR DESCRIPTION
**Pull request by marcoacierno**
## Description

Resolvers should receive the Strawberry info instead of the GraphQL Info

This means that the resolver info will receive the strawberry types instead of GraphQL types

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: strawberry/resolvers.py
Comment (BryceBeagle) [inline strawberry/resolvers.py:136]: Hmm I wonder if this should somehow be tied into our new [schema_converter](https://github.com/strawberry-graphql/strawberry/blob/update-resolver-info-type/strawberry/schema/schema_converter.py) system. The only complexion is that it's _from_ a `graphql-core` type, and not _to_.

Comment (marcoacierno) [reply strawberry/resolvers.py:136]: Ah that looks interesting

Comment (BryceBeagle) [reply strawberry/resolvers.py:136]: Our goal was to create an abstraction layer between the `graphql-core` code and the Strawberry backend

Comment (patrick91) [reply strawberry/resolvers.py:136]: @BryceBeagle can you think of a good place for that? 

we can also do this for now and then improve it later 😊

File: strawberry/types/info.py
Comment (patrick91) [inline strawberry/types/info.py:25]: do we need this? 🤔

File: strawberry/types/info.py
Comment (patrick91) [inline strawberry/types/info.py:20]: this will change to `StawberryType`, right @BryceBeagle? shall we add a comment for it?

File: strawberry/resolvers.py
Comment (patrick91) [inline strawberry/resolvers.py:135]: I'd remove this 😊
```suggestion
        original_info=info,
```

**Review by patrick91 — COMMENTED** [review]
Neat! Thanks for doing this!  Let's address the couple of comments and then we can merge this 😊

File: strawberry/resolvers.py
Comment (BryceBeagle) [reply strawberry/resolvers.py:136]: I was thinking about this, but I don't have any great answers yet. It may be good to have both `to` and `from` converters, but that's a lot of effort :sweat_smile:. How do we currently convert from graphql-core objects to Strawberry objects elsewhere in the codebase?

File: strawberry/types/info.py
Comment (BryceBeagle) [reply strawberry/types/info.py:20]: That's correct.

A comment is a good idea. We may also want to define a `TypeAlias` until `StrawberryType` is implemented.

File: strawberry/resolvers.py
Comment (patrick91) [reply strawberry/resolvers.py:136]: this is the only place where we convert graphql-core objects :)

Comment (BryceBeagle) [reply strawberry/resolvers.py:136]: Hmm, that certainly reduces motivation to create a whole abstracted system :thinking:

Comment (patrick91) [reply strawberry/resolvers.py:136]: yeah, I think we can figure out this later 😊 maybe we can have something in StrawberryResolver for this

File: strawberry/resolvers.py
Comment (marcoacierno) [reply strawberry/resolvers.py:135]: What, original_info?

File: strawberry/types/info.py
Comment (marcoacierno) [reply strawberry/types/info.py:25]: It depends, I guess until we expose all the information having the GraphQL info might be helpful?

File: strawberry/resolvers.py
Comment (patrick91) [reply strawberry/resolvers.py:135]: yes

File: strawberry/types/info.py
Comment (patrick91) [reply strawberry/types/info.py:25]: I'd just remove it for now and implement the remaining bits in another pr :)

File: strawberry/resolvers.py
Comment (marcoacierno) [reply strawberry/resolvers.py]: removed

File: strawberry/types/info.py
Comment (marcoacierno) [reply strawberry/types/info.py:21]: added a comment, is this what you meant?

File: RELEASE.md
Comment (patrick91) [inline RELEASE.md:1]: let's make this a minor 😊

```suggestion
Release type: minor
```

File: strawberry/types/info.py
Comment (patrick91) [reply strawberry/types/info.py:21]: yup!

File: RELEASE.md
Comment (patrick91) [reply RELEASE.md:1]: can you update the docs too? 😊 https://strawberry.rocks/docs/types/resolvers#accessing-execution-information

**Review by patrick91 — COMMENTED** [review]
Looks great I think we can merge this after updating the docs 😊

File: RELEASE.md
Comment (marcoacierno) [reply RELEASE.md]: done, let me know if the text (and type) makes sense in the docs

File: docs/types/resolvers.md
Comment (patrick91) [inline docs/types/resolvers.md:203]: is it optional?

Comment (marcoacierno) [reply docs/types/resolvers.md:203]: I copied the type from field_definition where it is optional, maybe it's wrong? 🤔

Comment (patrick91) [reply docs/types/resolvers.md:203]: actually, let's keep this hidden for now as we will change the type of `return_type` pretty soon

**Review by patrick91 — APPROVED** [review]
Let's get this merged 😊

**Pull request merged**
